### PR TITLE
feat(conf): Add board default pin configuration

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
This adds pinout locations for select boards.
The purpose is to allow a repository for how to wire the DUT to PHiLIP.
This can be used for CI exports or user exports.

_This reflects the pinout currently used in the HIL setup_

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
I guess I could use feedback, should I have empty fields to make boilerplate things easier or go for less clutter?

Should I add an export in front of everything?

Should it be a ?= instead of =

I wanted to preserve the *_GPIO names, the original idea was to put them in a comment but with it as a var we may want to use it in the future for a show pinout command, it contains the port/pin, the label on the board, and the connector location.

**Comment by MrKevinWeiss** [comment]
ping @smlng, maybe tomorrow we can get something decided since the SPI can really use this.

**Comment by MrKevinWeiss** [comment]
I still have a bit of a problem with application discovery.  It currently doesn't try to check the application...

**Comment by smlng** [comment]
put this into `Makefile.test_common`:

```
include $(TESTBASE)/dist/etc/conf/$(BOARD).env
```

**Comment by MrKevinWeiss** [comment]
I added a number of things, maybe we could split them.  I think everything should be prefixed with `HIL_` including the common things, this would make all env variables introduced specifically for these tests.

The question is should it be like that for `PHILIP_PORT` and `PORT`, should there be an abstraction (at some point we go `PORT=HIL_DUT_PORT`)?

I think with all this information we can actually generate a pinout for the boards that have the env setup.

Also I want to introduce the idea of `USE_HILDA` if we want to use a BPH+PI setup.  I am afraid this is going in the direction of CI driven development though.  It still has a purpose, for example when I ssh into a pi with the ancible setup I would want to just say `USE_HILDA=1 BOARD=<my_board> make flash robot-test -C <my_test>`, the `USE_HILDA` could also be done in the env setup though...  I guess so can all the parameters.  This may be useful later if we want to say, have the possibility of hard poweroff of the DUT or use any BPH commands.

Anyways lots to talk about.

File: tests/Makefile.tests_common [hunk 1]
```diff
@@ -4,17 +4,25 @@ APPLICATION ?= tests_$(notdir $(patsubst %/,%,$(CURDIR)))
...
 export CONNECT_WAIT
-# default periph/timer config
-PERIPH_TIMER_DEV ?= 0
-export PERIPH_TIMER_DEV
-PERIPH_TIMER_HZ ?= 1000000
-export PERIPH_TIMER_HZ
+
+HIL_RESET_WAIT ?= 3
+export HIL_RESET_WAIT
+RESET_WAIT ?= HIL_RESET_WAIT
```
Comment (smlng) [inline tests/Makefile.tests_common]: please remove these duplicate (old) variables. Looks like a work around to me, we should rather adapt the tests to use the new env variable names.

File: dist/etc/conf/arduino-due.env [hunk 2]
```diff
@@ -1,72 +1,77 @@
...
-RF_DEBUG1_PORT=1
-RF_DEBUG1_PIN=7
-
-RF_DEBUG2_GPIO="P27/13"
-RF_DEBUG2_PORT=1
-RF_DEBUG2_PIN=7
-
+HIL_DUT_RST_GPIO?=
+HIL_DUT_RST_PORT?=
+HIL_DUT_RST_PIN=
```
Comment (smlng) [inline dist/etc/conf/arduino-due.env]: why no `?=` for the pins? IIRC all should be defined that way to allow for overrides from command line

Comment (MrKevinWeiss) [reply dist/etc/conf/arduino-due.env]: up, missed it!

File: tests/Makefile.tests_common (hunk 1 above)
Comment (MrKevinWeiss) [reply tests/Makefile.tests_common]: OK, thought we would do a step by step but I can adapt everything.

**Comment by MrKevinWeiss** [comment]
Lets see what the CI catches.

**Comment by MrKevinWeiss** [comment]
It seems like the CI is happy. Should I squash?

File: tests/periph_timer/Makefile [hunk 3]
```diff
@@ -4,4 +4,14 @@ USEMODULE += shell
 FEATURES_REQUIRED += periph_timer
 FEATURES_REQUIRED += periph_gpio
 
+export HIL_DUT_GPIO0_PORT
+export HIL_DUT_GPIO0_PIN
+
+# default periph/timer config
+HIL_PERIPH_TIMER_DEV ?= 0
```
Comment (smlng) [inline tests/periph_timer/Makefile]: why? This should be in the .env file per board

File: tests/periph_timer/Makefile [hunk 4]
```diff
@@ -4,4 +4,14 @@ USEMODULE += shell
...
+export HIL_PERIPH_TIMER_DEV
+
+HIL_PERIPH_TIMER_HZ ?= 1000000
```
Comment (smlng) [inline tests/periph_timer/Makefile]: same here

File: dist/etc/conf/samr21-xpro.env [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+HIL_DUT_ADC_PORT?=
+HIL_DUT_ADC_PIN?=
+
+HIL_DUT_IC_GPIO?=
+HIL_DUT_IC_PORT?=
+HIL_DUT_IC_PIN?=
+
+HIL_DUT_GPIO0_GPIO?="PA19/EXT1-8"
+HIL_DUT_GPIO0_PORT?=0
+HIL_DUT_GPIO0_PIN?=19
```
Comment (smlng) [inline dist/etc/conf/samr21-xpro.env]: current config in Jenkins is debug0=pa13, debug1=pa18, and debug2=pa19 - and debug2 is okay on master but here all 3 fail now.

Comment (MrKevinWeiss) [reply dist/etc/conf/samr21-xpro.env]: Thanks, updated, we will see the CI

Comment (MrKevinWeiss) [reply dist/etc/conf/samr21-xpro.env]: It matches master now.  Anything else?

Comment (smlng) [reply dist/etc/conf/samr21-xpro.env]: nope, at least if you haven't mis-copied other pin configs to 😬

**Comment by MrKevinWeiss** [comment]
Thanks!

**Pull request merged**